src/Core/Settings.cpp

```diff
@@ -6027,6 +6027,14 @@ Enable PRQL - an alternative to SQL.
 Experimental tsToGrid aggregate function for Prometheus-like timeseries resampling. Cloud only
 )", EXPERIMENTAL) \
     \
+    DECLARE(UInt64, object_storage_max_nodes, 0, R"(
```

Comment:
I don't feel confident to review the whole PR, but if this is an experimental feature, than the related settings should start with `experimental_`, thus this should be `experimental_object_storage_max_nodes`.

Reply:
As I see it is not a strong requirement, some other experimental settings are named without `experimental` (like `live_view_heartbeat_interval`, `stop_refreshable_materialized_views_on_startup`, etc.).
May be better to mark setting as non-experimental?

Reply:
I think a lot of settings were introduced without mentioning experimental in their name as normal settings, and then marked as experimental in https://github.com/ClickHouse/ClickHouse/pull/71145 . So it is not a strong requirement, but it is better to adhere to it.

> May be better to mark setting as non-experimental?
I cannot answer this question as I miss a lot of context (the same reason why I am not confident reviewing the PR). If you decide to make this as non experimental, then we have to be more careful for sure.

Reply:
Make it non-experimental